src/hooks/useLatest.ts

```diff
@@ -0,0 +1,10 @@
+import { useRef, useEffect } from 'react';
+
+export function useLatest<T>(value: T) {
```

Comment:
Do we only use it for functions?
Maybe we could add a `useCallback` so we can skip the `current` part.